crates/holochain/src/conductor/conductor.rs

```diff
@@ -3298,6 +3308,102 @@ mod countersigning_impls {
+            if countersigning_workspace
+                .remove_countersigning_session()
+                .is_none()
+            {
+                // The session exists in the space as previously checked, so this case must never happen.
+                tracing::warn!(
+                    ?cell_id,
+                    "Could not remove countersigning session from workspace after abandoning it."
+                );
+            }
+            Ok(())
+        }
+
+        pub(crate) async fn publish_countersigning_session(
+            &self,
+            cell_id: &CellId,
+        ) -> ConductorResult<()> {
+            let space = self.get_or_create_space(cell_id.dna_hash())?;
+            let (countersigning_workspace, preflight_request) = self
+                .get_preflight_request_from_unknown_session(&space, cell_id)
+                .await?;
+            let cell = self.cell_by_id(cell_id).await?;
+            let network = Arc::new(cell.holochain_p2p_dna().clone());
+            force_publish_countersigning_session(
+                space,
+                network,
+                self.keystore().clone(),
+                cell.triggers().integrate_dht_ops.clone(),
+                cell.triggers().publish_dht_ops.clone(),
+                cell_id.clone(),
+                preflight_request,
+            )
+            .await?;
+            if countersigning_workspace
+                .remove_countersigning_session()
+                .is_none()
+            {
+                // The session exists in the space as previously checked, so this case must never happen.
+                tracing::warn!(
+                    ?cell_id,
+                    "Could not remove countersigning session from workspace after publishing it."
+                );
+            }
+            Ok(())
+        }
+
+        async fn get_preflight_request_from_unknown_session(
+            &self,
+            space: &Space,
+            cell_id: &CellId,
+        ) -> ConductorResult<(Arc<CountersigningWorkspace>, PreflightRequest)> {
+            let maybe_countersigning_workspace =
+                space.countersigning_workspaces.lock().get(cell_id).cloned();
+            match maybe_countersigning_workspace {
+                None => Err(ConductorError::CountersigningError(
+                    CountersigningError::WorkspaceDoesNotExist(cell_id.clone()),
+                )),
+                Some(countersigning_workspace) => {
+                    match countersigning_workspace.get_countersigning_session_state() {
+                        None => Err(ConductorError::CountersigningError(
+                            CountersigningError::SessionNotFound(cell_id.clone()),
+                        )),
+                        Some(CountersigningSessionState::Unknown {
+                            preflight_request, ..
+                        }) => Ok((countersigning_workspace, preflight_request)),
```

Comment:
The session will be placed in the `Unknown` state before the automatic resolution is run. We should check the inner state of the `Unknown` here to ensure that automatic resolution has run at least once. That just makes this very slightly safer for people. It'd be very easy to build a UI that shows these options too soon